src/darwin/Framework/CHIP/MTROTAProviderDelegateBridge.mm

```diff
@@ -0,0 +1,236 @@
+/**
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#import "MTROTAProviderDelegateBridge.h"
+
+#include <app/clusters/ota-provider/ota-provider.h>
+
+static NSInteger const kOtaProviderEndpoint = 0;
+
+MTROTAProviderDelegateBridge::MTROTAProviderDelegateBridge(void)
+    : mDelegate(nil)
+{
+}
+
+MTROTAProviderDelegateBridge::~MTROTAProviderDelegateBridge(void) {}
+
+void MTROTAProviderDelegateBridge::setDelegate(id<MTROTAProviderDelegate> delegate, dispatch_queue_t queue)
+{
+    if (delegate && queue) {
+        mDelegate = delegate;
+        mQueue = queue;
+    } else {
+        mDelegate = nil;
+        mQueue = nil;
+    }
+
+    chip::app::Clusters::OTAProvider::SetDelegate(kOtaProviderEndpoint, this);
+}
+
+void MTROTAProviderDelegateBridge::HandleQueryImage(chip::app::CommandHandler * commandObj,
+    const chip::app::ConcreteCommandPath & commandPath,
+    const chip::app::Clusters::OtaSoftwareUpdateProvider::Commands::QueryImage::DecodableType & commandData)
+{
+    // Make sure to hold on to the command handler and command path to be used in the completion block
+    __block chip::app::CommandHandler::Handle handle(commandObj);
+    __block chip::app::ConcreteCommandPath cachedCommandPath(
+        commandPath.mEndpointId, commandPath.mClusterId, commandPath.mCommandId);
+
+    id<MTROTAProviderDelegate> strongDelegate = mDelegate;
+    if ([strongDelegate respondsToSelector:@selector(handleQueryImage:completionHandler:)]) {
+        if (strongDelegate && mQueue) {
+            auto * commandParams = [[MTROtaSoftwareUpdateProviderClusterQueryImageParams alloc] init];
+            ConvertToQueryImageParams(commandData, commandParams);
+
+            dispatch_async(mQueue, ^{
+                [strongDelegate handleQueryImage:commandParams
```

Comment:
What does this do if `strongDelegate` does not implement this (optional, per the protocol definition) selector?  Or should it not be optional in the protocol, perhaps?

Reply:
Yeah, now that you bring it up, maybe this does not make sense to be optional.